flow-osgi/src/main/java/com/vaadin/flow/osgi/support/AppConfigFactoryTracker.java

```diff
@@ -0,0 +1,377 @@
+/**
+ * Copyright (C) 2020 Vaadin Ltd
+ *
+ * This program is available under Commercial Vaadin Developer License
+ * 4.0 (CVDLv4).
+ *
+ *
+ * For the full License, see <https://vaadin.com/license/cvdl-4.0>.
+ */
+package com.vaadin.flow.osgi.support;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletException;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Dictionary;
+import java.util.HashSet;
+import java.util.Hashtable;
+import java.util.Set;
+
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleEvent;
+import org.osgi.framework.BundleListener;
+import org.osgi.framework.FrameworkUtil;
+import org.osgi.framework.InvalidSyntaxException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.http.context.ServletContextHelper;
+import org.osgi.service.http.whiteboard.HttpWhiteboardConstants;
+import org.osgi.util.tracker.BundleTracker;
+import org.osgi.util.tracker.ServiceTracker;
+import org.slf4j.LoggerFactory;
+
+import com.vaadin.flow.di.Lookup;
+import com.vaadin.flow.internal.ApplicationClassLoaderAccess;
+import com.vaadin.flow.internal.VaadinContextInitializer;
+import com.vaadin.flow.osgi.support.OSGiVaadinInitialization.IllegalContextState;
+import com.vaadin.flow.osgi.support.OSGiVaadinInitialization.ResourceContextHelperFactory;
+import com.vaadin.flow.osgi.support.OSGiVaadinInitialization.ResourceService;
+import com.vaadin.flow.server.VaadinContext;
+import com.vaadin.flow.server.VaadinServlet;
+import com.vaadin.flow.server.VaadinServletContext;
+import com.vaadin.flow.server.startup.ApplicationConfiguration;
+import com.vaadin.flow.server.startup.ApplicationConfigurationFactory;
+
+class AppConfigFactoryTracker extends
```

Comment:
Internal implementation class which should have been `private`  nested but it's too big.

Do we really require javadocs for any class which is in its own file ?
At least SQ doesn't complain about it.

Docs are quite useless since it's OSGi where even `public`  class doesn't make open API necessary. This class can't be used outside of the bundle anyhow at all. 
But let's have impl details as javadocs.